### ISSUE TITLE
extend-immutable-calls does not checks imported function 

### ISSUE THREAD

**Issue by Tishka17** [PyCQA/flake8-bugbear#252]
I have small framework, which has `Depends` function simialr to fastapi feature. In my projects I want flake8-bubear to ignore only this funcion, but not any other function with same name. I've tried to set full link in flake config, but it does not work. 
I know that it is possible to always use full function name, but it looks dirty:

**Example 1.** Correct behavior - no B008
In config:
```
extend-immutable-calls=myapp.api.lambda_handler.depends.Depends
```
In code
```python
import myapp.api.lambda_handler.depends

@app.handler
def get_data(
        session: Session = myapp.api.lambda_handler.depends.Depends(),
) -> HttpResponse[ListResponse[AA]]:
```

**Example 2.** Incorrect behavior - B008
In config:
```
extend-immutable-calls=myapp.api.lambda_handler.depends.Depends
```
In code
```python
from myapp.api.lambda_handler.depends import Depends

@app.handler
def get_data(
        session: Session = Depends(),
) -> HttpResponse[ListResponse[AA]]:
```

**Example 3.** Incorrect behavior - no B008
```
extend-immutable-calls=Depends
```
```python
from myapp.api.lambda_handler.depends import Depends

@app.handler
def get_data(
        session: Session = Depends(),
) -> HttpResponse[ListResponse[AA]]:
    ...

def Depends(): 
    return []

@app.handler
def put_data(
        session: Session = Depends(),
) -> HttpResponse[ListResponse[AA]]:
```

**Comment by janrito** [comment]
Typer uses `typer.Optional`, and `typer.Argument` It would be nice to not have to disable B008

**Comment by cooperlees** [comment]
Sorry I never responded here. I don't totally understand what bugbear should do here. We're an AST parsing linter, so if someone can use the AST to be smarter here and not cause any regressions I'm open to this.

I am also open to adding gates with extra installs if needed - e.g. detect that typer is installed and take them into account. I don't know how feasible this is tho.

**Comment by janrito** [comment]
I don't really know what the right answer here is. There doesn't seem to be a way to annotate a function as immutable—doesn't look like there is going to be one anytime soon either https://github.com/python/typing/issues/646

Is there no way we can cannonicalise a method's name?

**Comment by 1678092075** [comment]
Hi! I can reproduce this on flake8-bugbear 25.11.29. I’d like to work on a focused fix that resolves imported callables before comparing them with `extend-immutable-calls`, while invalidating the imported binding if the name is later shadowed in the same scope. I’d add regression tests for `from` imports, aliases, and the redefinition case from Example 3. Does that direction fit the intended behavior?

**Comment by cooperlees** [comment]
This seems similar and would be a welcome addition. Thanks.

**Comment by abhnvgrg** [comment]
I'd like to take a look at this. Could a maintainer assign me?

My read of the current behaviour: B008's `extend-immutable-calls` matches on the callable's attribute path as written in the source, so it works when a function is called via its dotted path (e.g. `mod.func()`) but not when it's imported and called bare (`from mod import func` then `func()`), because the check doesn't resolve the import back to its module-qualified name.

Rough plan:

1. **Reproduce** with a minimal `flake8_bugbear` test case (imported callable in a default argument) to confirm behaviour matches the issue.
2. **Resolve the callable**: track `Import` / `ImportFrom` nodes in the visitor so an unqualified call can be matched against its module-qualified name in the user's `extend-immutable-calls` list. Handle `import ... as` aliases and `from ... import ... as` aliases.
3. **Preserve existing behaviour**: dotted-path matches keep working; only add the resolved-import path as an additional match, not a replacement.
4. **Tests**: add regression tests for direct import, aliased import, `from mod import name`, `from mod import name as alias`, and negative cases where a name shadows an import.

One thing I'll confirm against the existing visitor before starting: whether import tracking already exists somewhere in the codebase I can reuse rather than adding a parallel pass. Happy to adjust the approach if you'd rather solve this a different way.

**Comment by 1678092075** [comment]
Thanks, @abhnvgrg. I had already reproduced this and proposed the same scoped fix above, and @cooperlees confirmed that direction would be welcome. To avoid duplicating work, could a maintainer clarify who should proceed? I’m happy to coordinate or collaborate.

**Comment by abhnvgrg** [comment]
Thanks @1678092075 — you're clearly further along on this than I am, and with @cooperlees's directional nod already, it makes sense for you to run with it. I'll step back so there's no duplicated work. Happy to review or help test if that's useful once the PR is up.
